### DISCUSSION TITLE
SIP-5 (Creating JSON Web Tokens (JWTs)) Discussion

### DISCUSSION THREAD

**Discussion by Vid201** [MetaMask/SIPs#64] [SIPs]
A place to discuss SIP-5: Creating JSON Web Tokens (JWTs). The SIP proposes a new signing method - JWTs (one of the most used data standards on the internet).

**Comment by ritave** [comment]
As written in my initial review of the Pull Request:

From the perspective of the Snaps team, this functionality should be handled by the `snap_getBip44Entropy`.

On the other hand, this SIP has technical merits and so I merged this proposal on the editor's grounds. I'd like to see if any discussions breaks up concerning this change. If the community supports this change, it will be considered!

Keep in mind, that if no support from the community occurs, there's a high probability the Snaps team will not spend time to implement this change.

**Reply by Vid201** [reply]
@ritave, thanks for merging the SIP!

That is understandable; let's see if any other Snap also needs this functionality.

After some discussions, we decided to handle this functionality with `snap_getBip44Entropy` as your team suggested, which seems better than the deprecated `eth_sign`. I will also update SIP to mention the current approach for creating JWTs.
